### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by RahullRawat**
- Log in
- Log out
- Sign up

### CODE REVIEW

**Review by AnkurChunekar — APPROVED** [review]
- Add a logout button after user has logged in.
- Sign Up and Login Functionality works as intended

File: src/pages/Authentication/Login.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			username: "Guestuser",
+			password: "GuestUser123",
+		});
+	};
+
+	const logInHandler = async (e) => {
+		e.preventDefault();
+		if (userData.username && userData.password) {
+			const response = await dispatch(logIN(userData));
+			console.log(response);
```
Comment (AtulPant2704) [inline src/pages/Authentication/Login.js]: You can remove this console.

File: src/pages/Profile/Profile.js [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+import { logout } from "../../redux/slices/authSlice";
+import { toast } from "react-toastify";
 
 export const Profile = () => {
+	const dispatch = useDispatch();
+	const navigate = useNavigate();
+
+	const user = useSelector((state) => state.auth);
+
+	console.log(user);
```
Comment (AtulPant2704) [inline src/pages/Profile/Profile.js]: This console as well.

File: src/redux/thunks/authThunk.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export const logIN = createAsyncThunk(
+	"auth/login",
+	async (userData, { rejectWithValue }) => {
+		try {
+			const response = await axios.post("/api/auth/login", userData);
+			console.log("thunk", response);
+			const data = { data: response.data, status: response.status };
+			return data;
+		} catch (error) {
+			return rejectWithValue({ status: error.response.status });
```
Comment (AtulPant2704) [inline src/redux/thunks/authThunk.js]: You can send here status as well as the error that was sent from the backend.

**Review by AtulPant2704 — CHANGES_REQUESTED** [review]
- There are multiple console logs in your file, you can remove them.
- Whenever you are consoling an error write console.error(error) instead of console.log() so that your errors are separated from actual console.logs
- Your login is not working, may be you would have not added env variables to netlify. (Login is getting rejected with status 500)

**Pull request merged**